.github/workflows/ci.yml

```diff
@@ -97,6 +97,100 @@ jobs:
         with:
           path: /usr/local/cache/target
           key: cache-rust-artifacts-${{ hashFiles('Cargo.toml') }}
+  contract-size:
+    concurrency:
+      group: contract-size-${{ github.ref }}
+      cancel-in-progress: true
+    runs-on: ubuntu-latest
+    needs: examples-builds
+    container:
+      image: ghcr.io/supercolony-net/openbrush-contracts-ci
+      options: --user root
+      env:
+        CARGO_TARGET_DIR: /usr/local/cache/target
+    steps:
+      - name: Cache contract artifacts
+        id: cache-contract-artifacts
+        uses: actions/cache@v3
+        with:
+          path: artifacts
+          key: cache-contract-artifacts-${{ github.sha }}
+
+      - name: Find contract data
+        id: find-data
+        run: |
+          MAIN_DATA=$(find . -type f -name "*.contract" -printf '%s ' -exec basename \{} .contract \; | sort -k2)
+          echo "MAIN_DATA<<EOF" >> $GITHUB_ENV
+          echo "$MAIN_DATA" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Yarn install dependencies
+        run: |
+          yarn
+          yarn add ts-node
+
+      - name: Cache rust artifacts for PR branch
+        id: cache-rust-artifacts-pr
+        uses: actions/cache@v3
+        with:
+          path: /usr/local/cache/target
+          key: cache-rust-artifacts-pr-${{ hashFiles('Cargo.toml') }}-${{ github.ref }}
+          restore-keys: |
+            cache-rust-artifacts-pr-${{ hashFiles('Cargo.toml') }}
+
+      - name: Cache contract artifacts for PR branch
+        id: cache-contract-artifacts-pr
+        uses: actions/cache@v3
+        with:
+          path: artifacts
+          key: cache-contract-artifacts-pr-${{ github.sha }}
+
+      - name: Redspot Сompile examples for PR branch
```

Comment:
You can reuse artifacts generated for integration tests. We already have a separate build phase that posts artifacts.